x/wasm/internal/keeper/cosmwasm/msg.go

```diff
@@ -0,0 +1,56 @@
+package cosmwasm
+
+import (
+	"encoding/json"
+
+	cosmwasmv1 "github.com/CosmWasm/go-cosmwasm/types"
+)
+
+// CosmosMsg is an rust enum and only (exactly) one of the fields should be set
+// Should we do a cleaner approach in Go? (type/data?)
+type CosmosMsg struct {
+	Bank    *cosmwasmv1.BankMsg    `json:"bank,omitempty"`
+	Custom  json.RawMessage        `json:"custom,omitempty"`
+	Staking *cosmwasmv1.StakingMsg `json:"staking,omitempty"`
+	Wasm    *cosmwasmv1.WasmMsg    `json:"wasm,omitempty"`
+	IBC     *IBCMsg                `json:"wasm_ibc,omitempty"`
+}
+
+type IBCMsg struct {
+	SendPacket   *IBCSendMsg         `json:"execute,omitempty"`
+	CloseChannel *IBCCloseChannelMsg `json:"instantiate,omitempty"`
+	Transfer     *IBCTransferMsg     `json:"instantiate,omitempty"`
```

Comment:
Transfer is for the ics20 / ibc-transfer module? Or something else?

Can you add a comment there for the purpose? SendMsg and CloseChannel are clear. And if you want to add in OpenChannel you can (I decided you were right and just put that responsibility back to the relayer)